crates/gateway-types/src/reply.rs

```diff
@@ -346,11 +350,54 @@ pub mod transaction {
         }
     }
 
+    #[derive(Clone, Debug, Serialize, PartialEq, Eq)]
+    #[serde(tag = "version")]
+    pub enum DeclareTransaction {
+        #[serde(rename = "0x0")]
+        V0(DeclareTransactionV0),
+        #[serde(rename = "0x1")]
+        V1(DeclareTransactionV1),
```

Comment:
Hmm, I think we already have DECLARE v1 and we should be adding DECLARE v2? (Check [this testnet transaction](https://alpha4.starknet.io/feeder_gateway/get_transaction?transactionHash=0x04f804f16a8512ccdd50161fdd77c4ff3662a96218621a78a9212095456b7268))

Reply:
Where is DECLARE v0 then? (because that [does](https://alpha4.starknet.io/feeder_gateway/get_transaction?transactionHash=0x6d346ba207eb124355960c19c737698ad37a3c920a588b741e0130ff5bd4d6d)also exist..).

Or does that mean that v0 and v1 are identical, just with different version fields allowed?

Reply:
We had a single type to handle both versions. Handled by adding V0, V1, and now V2 variants.